google/resource_container_node_pool.go

```diff
@@ -157,14 +164,36 @@ func resourceContainerNodePoolCreate(d *schema.ResourceData, meta interface{}) e
 
 	mutexKV.Lock(containerClusterMutexKey(project, zone, cluster))
 	defer mutexKV.Unlock(containerClusterMutexKey(project, zone, cluster))
-	op, err := config.clientContainer.Projects.Zones.Clusters.NodePools.Create(project, zone, cluster, req).Do()
+	var op interface{}
+	switch containerApiVersion {
+	case v1:
+		reqV1 := &container.CreateNodePoolRequest{}
+		err = Convert(req, reqV1)
+		if err != nil {
+			return err
+		}
 
-	if err != nil {
-		return fmt.Errorf("Error creating NodePool: %s", err)
+		op, err = config.clientContainer.Projects.Zones.Clusters.NodePools.Create(project, zone, cluster, reqV1).Do()
+		if err != nil {
+			return fmt.Errorf("Error creating NodePool: %s", err)
+		}
+	case v1beta1:
+		reqV1Beta := &containerBeta.CreateNodePoolRequest{}
+		err = Convert(req, reqV1Beta)
+		if err != nil {
+			return err
+		}
+
+		op, err = config.clientContainerBeta.Projects.Zones.Clusters.NodePools.Create(project, zone, cluster, reqV1Beta).Do()
+		if err != nil {
+			return fmt.Errorf("Error creating NodePool: %s", err)
+		}
 	}
 
+	d.SetId(fmt.Sprintf("%s/%s/%s", zone, cluster, nodePool.Name))
```

Comment:
Consider including this change in a separate CL because it is not strictly related to beta scafolding